src/cascadia/TerminalApp/CascadiaSettingsSerialization.cpp

```diff
@@ -262,13 +262,45 @@ void CascadiaSettings::_WriteSettings(const std::string_view content)
 std::optional<std::string> CascadiaSettings::_ReadSettings()
 {
     auto pathToSettingsFile{ CascadiaSettings::GetSettingsPath() };
-    const auto hFile = CreateFileW(pathToSettingsFile.c_str(), GENERIC_READ, FILE_SHARE_READ | FILE_SHARE_WRITE, NULL, OPEN_EXISTING, FILE_ATTRIBUTE_NORMAL, NULL);
+    auto hFile = CreateFileW(pathToSettingsFile.c_str(), GENERIC_READ, FILE_SHARE_READ | FILE_SHARE_WRITE, NULL, OPEN_EXISTING, FILE_ATTRIBUTE_NORMAL, NULL);
     if (hFile == INVALID_HANDLE_VALUE)
     {
-        // If the file doesn't exist, that's fine. Just log the error and return
-        //      nullopt - we'll create the defaults.
-        LOG_LAST_ERROR();
-        return std::nullopt;
+        // GH#1770 - Now that we're _not_ roaming our settings, do a quick check
+        // to see if there's a file in the Roaming App data folder. If there is
+        // a file there, but not in the LocalAppData, it's likely the user is
+        // upgrading from a version of the terminal from before this change.
+        // We'll try moving the file from the Roaming app data folder to the
+        // local appdata folder.
+
+        auto pathToRoamingSettingsFile{ CascadiaSettings::_GetRoamingSettingsPath() };
+        const auto hRoamingFile = CreateFileW(pathToRoamingSettingsFile.c_str(), GENERIC_READ, FILE_SHARE_READ | FILE_SHARE_WRITE, NULL, OPEN_EXISTING, FILE_ATTRIBUTE_NORMAL, NULL);

```

Comment:
we may want to open this with a different sharing mode to make sure that two instances started at the exact same time don't try to move the file

Reply:
My Win32-foo is a little rusty here (and I know this is fragile code).

Would I just remove the `FILE_SHARE_READ | FILE_SHARE_WRITE` here? That would make it so only this instance could open that file, yea? But then what would happen if two are opening at the same time, the first opens the file to move it, then the second finds that it can't open the file, and proceeds to open with the defaults (since neither the local or roaming file could be opened). That doesn't seem right to me - how would I get around that?

Reply:
meh, it may not be worth doing

Reply:
I'm just leaving a note in the code. If it ever happens in the wild, we can revisit.